domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs

```diff
@@ -140,15 +148,20 @@ public async Task SeedConnectionInvitations(IEnumerable<ConnectionInvitation> in
         await dbContext.SaveChangesAsync();
     }
 
-    public HttpClient CreateUnauthenticatedClient() => CreateClient();
+    public HttpClient CreateUnauthenticatedClient()
+    {
+        var client = CreateClient();
+        client.DefaultRequestHeaders.Add("EO_API_VERSION", "20230101");
+        return client;
+    }
 
     public HttpClient CreateAuthenticatedClient(string sub, string tin = "11223344", string name = "Peter Producent",
         string actor = "d4f32241-442c-4043-8795-a4e6bf574e7f")
     {
         var client = CreateClient();
         client.DefaultRequestHeaders.Authorization =
             new AuthenticationHeaderValue("Bearer", GenerateToken(sub: sub, tin: tin, name: name, actor: actor));
-
+        client.DefaultRequestHeaders.Add("EO_API_VERSION", "20230101");
```

Comment:
I suggest to have an optional parameter in `CreateAuthenticatedClient` to set the API version.

Reply:
fixed